internal/pkg/override/patch.go

```diff
@@ -0,0 +1,283 @@
+				return errStopFollowingPointer
+			}
+
+			idx, err := strconv.Atoi(pointer[0])
+			switch {
+			case err != nil:
+				return fmt.Errorf("expected index in sequence, got %q", pointer[0])
+			case idx < 0 || idx > len(node.Content):
+				return fmt.Errorf("invalid index %d for sequence of length %d", idx, len(node.Content))
+			}
+
+			// add node at idx
+			node.Content = append(node.Content[:idx], append([]*yaml.Node{&p.Value}, node.Content[idx:]...)...)
+			return errStopFollowingPointer
+		}
+
+		return nil
+	})
+}
+
+func (p *yamlPatch) applyRemove(root *yaml.Node) error {
+	return followJSONPointer(root, p.Path, func(node *yaml.Node, pointer []string) error {
+		if len(pointer) != 1 {
+			return nil
+		}
+
+		switch node.Kind {
+		case yaml.DocumentNode:
+			// make sure we are encoding zero into node
+			p.Value = yaml.Node{}
+			return p.encodeAndStop(node)
+		case yaml.MappingNode:
+			for i := 0; i < len(node.Content); i += 2 {
+				if node.Content[i].Value == pointer[0] {
+					node.Content = append(node.Content[:i], node.Content[i+2:]...)
+					return errStopFollowingPointer
+				}
+			}
+		case yaml.SequenceNode:
+			idx, err := strconv.Atoi(pointer[0])
+			switch {
+			case err != nil:
+				return fmt.Errorf("expected index in sequence, got %q", pointer[0])
+			case idx < 0 || idx > len(node.Content)-1:
+				return fmt.Errorf("invalid index %d for sequence of length %d", idx, len(node.Content))
+			}
+
+			node.Content = append(node.Content[:idx], node.Content[idx+1:]...)
+			return errStopFollowingPointer
+		}
+
+		return nil
+	})
+}
+
+func (p *yamlPatch) applyReplace(root *yaml.Node) error {
+	if p.Value.IsZero() {
+		return fmt.Errorf("value required")
+	}
+
+	return followJSONPointer(root, p.Path, func(node *yaml.Node, pointer []string) error {
+		if len(pointer) > 0 {
+			return nil
+		}
+		return p.encodeAndStop(node)
+	})
+}
+
+func (p *yamlPatch) encodeAndStop(node *yaml.Node) error {
+	if err := node.Encode(p.Value); err != nil {
+		return err
+	}
+	return errStopFollowingPointer
+}
+
+var errStopFollowingPointer = errors.New("stop following pointer")
+
+type visitNodeFunc func(node *yaml.Node, remaining []string) error
+
+func followJSONPointer(root *yaml.Node, pointer string, visit visitNodeFunc) error {
+	split := strings.Split(pointer, "/")
+	for i := range split {
+		// apply replacements as described https://www.rfc-editor.org/rfc/rfc6901#section-4
+		split[i] = strings.ReplaceAll(split[i], "~1", "/")
+		split[i] = strings.ReplaceAll(split[i], "~0", "~")
+	}
+
+	return followJSONPointerHelper(root, nil, split, visit)
+}
+
+func followJSONPointerHelper(node *yaml.Node, traversed, remaining []string, visit visitNodeFunc) error {
+	if err := visit(node, remaining); err != nil {
+		if errors.Is(err, errStopFollowingPointer) {
+			return nil
+		}
+		return fmt.Errorf("key %q: %w", strings.Join(traversed, jsonPointerSeparator), err)
+	} else if len(remaining) == 0 {
+		return nil
+	}
```

Comment:
nit: this seems to be the same?
```suggestion
	}
	if len(remaining) == 0 {
		return nil
	}
```

Reply:
code is gone in refactor!